specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json

```diff
@@ -5462,6 +5492,43 @@
       ],
       "description": "Describes a Virtual Machine."
     },
+    "VirtualMachineUpdate": {
+      "properties": {
+        "plan": {
+          "$ref": "#/definitions/Plan",
+          "description": "Specifies information about the marketplace image used to create the virtual machine. This element is only used for marketplace images. Before you can use a marketplace image from an API, you must enable the image for programmatic use.  In the Azure portal, find the marketplace image that you want to use and then click **Want to deploy programmatically, Get Started ->**. Enter any required information and then click **Save**."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VirtualMachineProperties"
+        },
+        "resources": {
+          "readOnly": true,
```

Comment:
what does it mean for a property of `VirtualMachineUpdate` (if I see correctly, only ever sent) to be `readOnly`? 😃

Reply:
Yup, you're right. It makes no sense, so I removed it. Originally, I copied that from the VirtualMachine object, which is used in GETs as well, so it makes sense to have it in that object as readOnly, but it doesn't make sense for the VirtualMachineUpdate object :)